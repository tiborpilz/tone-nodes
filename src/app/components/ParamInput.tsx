'use client';

import { useState } from 'react';
import { Param } from 'tone';
import { UnitName } from 'tone/build/esm/core/type/Units';

export default function ParamInput<T extends UnitName>({
  param,
  label,
}: {
  param: Param<T>,
  label: string,
}) {
  const [value, setValue] = useState(param.value as number);

  return (
    <label>
      <span>{label}</span>
      <input
        type="range"
        value={value}
        onChange={(e) => {
          setValue(parseFloat(e.target.value));
          param.value = parseFloat(e.target.value);
        }}
      />
      <span>{value}</span>
    </label>
  );
}
