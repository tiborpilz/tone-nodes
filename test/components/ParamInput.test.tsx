import { render } from '@testing-library/react';
import { Param } from 'tone';
import ParamInput from '@/app/components/ParamInput';
import { describe, expect, it, mock } from 'bun:test';

const invokeMock = mock();

mock.module('@tauri-apps/api/core', () => ({
  invoke: invokeMock,
}));

const listenMock = mock();

mock.module('@tauri-apps/api/event', () => ({
  listen: listenMock,
}));

const mockAudioParam = {
  value: 0,
} as unknown as Param;

describe('ParamInput', () => {
  it('should render without crashing', () => {
    render(<ParamInput param={mockAudioParam} label="Mock Param" />);

    const component = document.querySelector('[data-testid="param-input"]');
    expect(component?.innerHTML).toMatchSnapshot();
  });
});
