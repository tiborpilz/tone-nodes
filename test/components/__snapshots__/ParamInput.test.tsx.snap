// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`ParamInput should render without crashing 1`] = `"<span>Mock Param</span><span>not learning</span><input type="range" min="0" max="1" step="0.01" value="0"><span>0</span><button>Learn MIDI</button>"`;
